shared/actions/chat/attachment.js

```diff
@@ -167,8 +167,20 @@ function* onLoadAttachment({
     }
   }
 
+  function _tmpFileName(
```

Comment:
I moved some things out of shared into the sagas if they're really not shared at all